modules/slate-sugar/src/main/ts/ephox/slate-sugar/api/Api.ts

```diff
@@ -0,0 +1,79 @@
+import { Maybe } from '@ephox/katamari';
+
+type path = number[];
+
+interface Editor {}
+
+interface RtcAdaptor {
+  raw: {
+    getModel: () => Maybe<any>;
+    getModelApi: () => Maybe<ModelApi>;
+  };
+}
+
+interface RtcEditor extends Editor {
+  rtcInstance: RtcAdaptor;
+}
+
+interface SlateElement {
+  type: string;
+  children: Array<SlateNode>;
+  class?: string;
+}
+
+interface SlateText {
+  text: string;
+}
+
+type SlateNode = SlateElement | SlateText;
+
+interface List<T>{
+  hd: T;
+  tl: List<T> | 0;
+}
+
+interface SlateLoc {
```

Comment:
Do we have any ideas on what to call this I think all references to slate should be removed on the external api. Maybe we should just call it Location but maybe that clashes with dom interfaces or ModelLocation or ModelLoc or just Loc naming things are hard?

Reply:
Can confirm `Location` is in the native DOM types. `ModelLocation` probably makes the most sense to me, as it's still readable/clear what it's applicable to.

Reply:
renamed Slate references to Model